_posts/en/newsletters/2025-05-16-newsletter.md

```diff
@@ -0,0 +1,150 @@
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
+[Lightning BLIPs][blips repo], [Bitcoin Inquisition][bitcoin inquisition
+repo], and [BINANAs][binana repo]._
+
+- [Bitcoin Core #32155][] updates the internal miner to [timelock][topic
+  timelocks] coinbase transactions by setting the `nLockTime` field to the
+  current block’s height minus one and requiring the `nSequence` field not to be
+  final (to enforce the timelock). Although the built-in miner isn’t typically
+  used on mainnet, updating it encourages mining pools to adopt these changes
+  early in their own software in preparation for the proposed [BIP54][]
+  [consensus cleanup]​[topic consensus cleanup] soft fork. Timelocking coinbase
+  transactions solves the [duplicate transaction][topic duplicate transactions]
+  vulnerability, and would allow the costly [BIP30][] checks to be lifted.
+
+- [Bitcoin Core #28710][] removes the remaining legacy wallet code,
+  documentation, and related tests. This includes the legacy-only RPCs, such as
+  `importmulti`, `sethdseed`, `addmultisigaddress`, `importaddress`,
+  `importpubkey`, `dumpwallet`, `importwallet`, and `newkeypool`. As the final
+  step for legacy wallet removal, the BerkeleyDB dependency and related
+  functions are also removed. However, the bare minimum of legacy code and an
+  independent BDB parser (see Newsletter [#305][news305 bdb]) are retained in
+  order to perform wallet migration to [descriptor][topic descriptors] wallets.
+
+- [Core Lightning #8272][] disables the DNS seed lookup peer discovery fallback
+  from the connection daemon `connectd` to resolve call block issues caused by
+  offline DNS seeds.
+
+- [LND #8330][] adds a small constant (1/c) to the pathfinding bimodal
+  probability model to address numerical instability. In edge cases where the
+  calculation would otherwise fail due to rounding errors and produce a zero
+  probability, this regularization provides a fallback by causing the model to
+  revert to a uniform distribution. This resolves normalization bugs that occur
+  in scenarios involving very large channels or channels that don't fit a
+  bimodal distribution. Additionally, the model now skips unnecessary
+  probability calculations and automatically corrects outdated channel liquidity
+  observations and contradictory historical information.
+
+- [Rust Bitcoin #4458][] replaces the `MtpAndHeight` struct with an explicit
+  pair of the newly added `BlockMtp` and the already existing `BlockHeight`,
+  enabling better modeling of both block height and Median Time Past (MTP)
+  values in relative [timelocks][topic timelocks]. Unlike
+  `locktime::absolute::Mtp`, which is constrained to values above 500 million
```

Comment:
```suggestion
  `locktime::absolute::MedianTimePast`, which is constrained to values above 500 million
```

It was later [renamed](https://github.com/rust-bitcoin/rust-bitcoin/pull/4462/files#diff-4cd8e9b2bdc17689f9bf7e9d43908beb4c8d2c87c4752903f23e783915af9806).

Reply:
Thanks for this keen observation!  Updating.